src/selection_criteria.rs

```diff
@@ -84,7 +86,7 @@ pub type Predicate = Arc<dyn Send + Sync + Fn(&ServerInfo) -> bool>;
 /// option and will be sent to the server as an integer number of seconds.
 ///
 /// See the [MongoDB docs](https://docs.mongodb.com/manual/core/read-preference) for more details.
-#[derive(Clone, Debug, PartialEq)]
+#[derive(Clone, Debug, Deserialize, PartialEq)]
```

Comment:
This also won't work, but for more frustrating reasons. According to the [server selection specification](https://github.com/mongodb/specifications/blob/master/source/server-selection/server-selection.rst#read-preference), a read preference is supposed to look like this:
```json
"readPreference": { "mode": "primaryPreferred", "tag_sets": [] }
```
In the [spec tests](https://github.com/mongodb/specifications/blob/master/source/transactions/tests/transaction-options.json#L1102) however, sometimes what's actually provided is just a string that only contains the mode, and [in other cases](https://github.com/mongodb/specifications/blob/master/source/transactions/tests/read-pref.json#L114), the correct format is supplied. Ideally, the specs would respect the correct shape or at least be consistent in the representation they do use, but unfortunately neither is the case. To resolve this issue, I think you'll need to implement a custom `Deserialize` implementation for `ReadPreference`, one that first attempts to deserialize a string and, failing that, attempts to deserialize a struct with the `mode`, `tag_sets`, and `maxStalenessSeconds` fields per the spec.

Given that all of this is only necessary for the transactions runner as far as I can tell, I think leaving a TODO with a follow up ticket would be sufficient. What do you think @saghm ?

Reply:
Yeah, I'm fine with leaving this until later

Reply:
https://jira.mongodb.org/browse/RUST-495